src/components/Search/Search.js

```diff
@@ -223,46 +246,58 @@ const Search = () => {
   }
 
   // remove old footprints from map
-  function removeFootprints() {
-    resultFootprints.eachLayer(function (layer) {
-      map.removeLayer(layer)
+  function removeFootprints(resultFootPrintsLocal) {
+    resultFootPrintsLocal.eachLayer(function (layer) {
+      resultFootPrintsLocal.removeLayer(layer)
     })
   }
 
-  // function called when search button clicked
-  function searchAPI() {
-    // get viewport bounds
-    const viewportBounds = map.getBounds()
+  // throttle function to prevent map from rendering too quickly
+  const debounce = (func, wait) => {
+    let timeout
+    return function executedFunction(...args) {
+      const later = () => {
+        clearTimeout(timeout)
+        func(...args)
+      }
+      clearTimeout(timeout)
+      timeout = setTimeout(later, wait)
+    }
+  }
+
+  // search throttle set to 1000ms
+  const processSearch = debounce(function (resultFootprintsInit) {
+    searchAPI(resultFootprintsInit)
+  }, 1000)
```

Comment:
I think the lambda definition here is unnecessary and you can just do:

```suggestion
  const processSearch = debounce(searchAPI, 1000)
```

Reply:
This has been changed to: 
`const processSearch = (resultFootprintsInit) => debounce(searchAPI(resultFootprintsInit), 1000)`